core/src/datayoga_core/block.py

```diff
@@ -93,6 +95,22 @@ def stop(self):
     def get_block_name(self):
         return os.path.basename(os.path.dirname(sys.modules[self.__module__].__file__))
 
+    def produce(self) -> Generator[Message, None, None]:
+        """Produces data
+
+        Returns:
+            Generator[Message]: Produced data
+        """
+        raise NotImplementedError
+
+    def ack(self, msg_ids: List[str]):
```

Comment:
these don't belong in Block. Producer currently inherits from Block and adds those.

Reply:
Yep. But it's breaks typing. It's because I move it. Will be completely solved when we split Block and Producer.

Reply:
We can't put in code that doesn't belong here because linter or typing doesn't agree. that doesn't make sense.